File: VantaSpeech-macOS/VantaSpeech/Models/Recording.swift
import Foundation
import SwiftData

@Model
final class Recording {
    var id: UUID
    var title: String
    var createdAt: Date
    var duration: TimeInterval
    var audioFileURL: String
    var transcriptionText: String?
    var summaryText: String?
    var isTranscribed: Bool
    var isUploading: Bool

    init(
        id: UUID = UUID(),
        title: String,
        createdAt: Date = Date(),
        duration: TimeInterval = 0,
        audioFileURL: String,
        transcriptionText: String? = nil,
        summaryText: String? = nil,
        isTranscribed: Bool = false,
        isUploading: Bool = false
    ) {
        self.id = id
        self.title = title
        self.createdAt = createdAt
        self.duration = duration
        self.audioFileURL = audioFileURL
        self.transcriptionText = transcriptionText
        self.summaryText = summaryText
        self.isTranscribed = isTranscribed
        self.isUploading = isUploading
    }
}

extension Recording {
    var formattedDuration: String {
        let hours = Int(duration) / 3600
        let minutes = (Int(duration) % 3600) / 60
        let seconds = Int(duration) % 60

        if hours > 0 {
            return String(format: "%d:%02d:%02d", hours, minutes, seconds)
        } else {
            return String(format: "%d:%02d", minutes, seconds)
        }
    }

    var formattedDate: String {
        let calendar = Calendar.current
        let now = Date()

        if calendar.isDateInToday(createdAt) {
            let formatter = DateFormatter()
            formatter.dateFormat = "HH:mm"
            return "Today, \(formatter.string(from: createdAt))"
        } else if calendar.isDateInYesterday(createdAt) {
            let formatter = DateFormatter()
            formatter.dateFormat = "HH:mm"
            return "Yesterday, \(formatter.string(from: createdAt))"
        } else {
            let formatter = DateFormatter()
            formatter.dateFormat = "MMM d, yyyy"
            return formatter.string(from: createdAt)
        }
    }

    var transcriptionPreview: String? {
        guard let text = transcriptionText else { return nil }
        if text.count > 150 {
            return String(text.prefix(150)) + "..."
        }
        return text
    }
}

enum AudioQuality: String, CaseIterable, Identifiable {
    case low = "Low"
    case medium = "Medium"
    case high = "High"

    var id: String { rawValue }

    var bitrate: Int {
        switch self {
        case .low: return 64
        case .medium: return 96
        case .high: return 128
        }
    }

    var label: String {
        switch self {
        case .low: return "Low (64 kbps)"
        case .medium: return "Medium (96 kbps)"
        case .high: return "High (128 kbps)"
        }
    }
}

File: VantaSpeech-Android/app/src/main/java/com/vantaspeech/data/model/Recording.kt
package com.vantaspeech.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.Instant
import java.time.LocalDateTime
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.util.UUID
import kotlin.time.Duration
import kotlin.time.Duration.Companion.seconds

@Entity(tableName = "recordings")
data class Recording(
    @PrimaryKey
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val createdAt: Long = System.currentTimeMillis(),
    val duration: Long = 0, // Duration in seconds
    val audioFilePath: String,
    val transcriptionText: String? = null,
    val summaryText: String? = null,
    val isTranscribed: Boolean = false,
    val isUploading: Boolean = false
) {
    val formattedDuration: String
        get() {
            val durationSeconds = duration.seconds
            val hours = durationSeconds.inWholeHours
            val minutes = (durationSeconds.inWholeMinutes % 60)
            val seconds = (durationSeconds.inWholeSeconds % 60)

            return if (hours > 0) {
                String.format("%d:%02d:%02d", hours, minutes, seconds)
            } else {
                String.format("%d:%02d", minutes, seconds)
            }
        }

    val formattedDate: String
        get() {
            val instant = Instant.ofEpochMilli(createdAt)
            val localDateTime = LocalDateTime.ofInstant(instant, ZoneId.systemDefault())
            val now = LocalDateTime.now()

            return when {
                localDateTime.toLocalDate() == now.toLocalDate() -> {
                    "Today, ${localDateTime.format(DateTimeFormatter.ofPattern("HH:mm"))}"
                }
                localDateTime.toLocalDate() == now.minusDays(1).toLocalDate() -> {
                    "Yesterday, ${localDateTime.format(DateTimeFormatter.ofPattern("HH:mm"))}"
                }
                else -> {
                    localDateTime.format(DateTimeFormatter.ofPattern("MMM d, yyyy"))
                }
            }
        }

    val transcriptionPreview: String?
        get() = transcriptionText?.take(150)?.let {
            if (transcriptionText.length > 150) "$it..." else it
        }
}

enum class AudioQuality(val bitrate: Int, val label: String) {
    LOW(64, "Low (64 kbps)"),
    MEDIUM(96, "Medium (96 kbps)"),
    HIGH(128, "High (128 kbps)")
}
